lib/cli/cmd-add.js

```diff
@@ -0,0 +1,23 @@
+var Q = require('q');
+var runBinary = require('./run-binary');
+
+// `donejs add app [folder]`     => `donejs-cli init [folder]`
+// `donejs add plugin [folder]`  => `donejs-cli init [folder] --type=plugin`
+// `donejs add generator [name]` => `donejs-cli init [folder] --type=generator`
+module.exports = function(type, folder, options) {
+  if (!isValidType(type)) {
+    var msg = 'Invalid [type] value `' + type + '`';
+    return Q.reject(new Error(msg));
+  }
+
+  if (type !== 'app') {
+    options.type = type;
+  }
+
+  return runBinary(['init', folder], options);
+};
+
+function isValidType(type) {
```

Comment:
See, I like this style for that reason, if you have a really long function I like seeing the `return` early in the function body, I think it makes it easier to understand how it flows. Like:

```js
function doSomeStuff(){
  var a = 'foo';
  var b = bar';

  return doOtherStuff();

  function doOtherStuff() {
    ... this is a super long function
  }
}
```